presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/optimization/JdbcSql.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.jdbc.optimization;
+
+import com.facebook.presto.spi.relation.ConstantExpression;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+public class JdbcSql
+{
+    private final String sql;
+    private final List<ConstantExpression> bindValues;
+
+    public JdbcSql(String sql)
+    {
+        this(sql, ImmutableList.of());
+    }
+
+    public JdbcSql(String sql, List<ConstantExpression> bindValues)
+    {
+        this.sql = sql;
+        this.bindValues = bindValues;
+    }
+
+    public String getSql()
+    {
+        return sql;
+    }
+
+    public List<ConstantExpression> getBindValues()
```

Comment:
What is this for?

Reply:
It is for when the table layout handle is used to execute the query, we can create a PreparedStatement and use the bind values as parameters to the PreparedStatement. I.e. all constants are translated as "?" and then their corresponding actual value is added to the bindValues list. Without doing this we would lose precision on double/float types etc.

Reply:
Adding Javadoc since that isn't clear from reading the code.